### ISSUE TITLE
Update BoNeMEAL for BanManager v5

### ISSUE THREAD

**Issue by ftbastler** [ftbastler/BoNeMEAL#66]
- [x] Update Laravel to v5.1
- [ ] Integrate #63
- [x] Integrate #62
- [ ] Integrate player reports ("bm_player_reports" table)
- [ ] Integrate IP range bans ("bm_ip_range_bans" and "bm_ip_range_ban_records" tables)
- [ ] Improve the general look and feel

**Comment by ftbastler** [comment]
@confuser With v5 the database structure changed quite a bit. There are some new tables and some new rows which I don't know what they do.

I need some info about..
- IP range bans
- The column "soft"
- The column "createdReason"
- The table bm_player_reports (I could not find a cmd for that)
- The table warnings (particularly the columns "expires" and "read")

**Comment by confuser** [comment]
- Similar implementation to ip bans, uses fromIp and toIp as longs (Use INET_ATON in MySQL to convert)
- `soft` is for shadow mutes, e.g. players can send messages in chat, but nobody else sees them
- `createdReason` is for unban/unmute reasons in the record tables
- `bm_player_reports` is part of /bmreport, however it's not yet fully implemented/finished
- `expires` is the timestamp for when the warning expires, `read` is if the player has seen the warning or not

**Comment by ftbastler** [comment]
@confuser A warning can expire? So what happens when it does that/whats the difference from v4 warnings?

**Comment by confuser** [comment]
Yes, it can expire. When it does, it's deleted from the table, there's no records table, so it's completely gone.

**Comment by ftbastler** [comment]
Ah, okay. That is quite interesting.

**Comment by oranges13** [comment]
Hey @confuser I am working on trying to pick up the issues on this repo. The `bm_player_reports_commands` table is the only one I'm confused about. How do you think that would fit into the interface? Thanks for any info you can provide.
